dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/cache/CacheEntry.java

```diff
@@ -240,42 +253,49 @@ public void write() throws IOException {
                 if (fileEntries.isEmpty())
                     Tr.debug(this, tc, "Not writing cache entry, fileEntries is empty");
             }
-            return;
+            return CompletableFuture.completedFuture(null);
         }
 
         if (LoggingUtils.isDebugEnabled(tc)) {
-            Tr.debug(this, tc, "Writing cache entry");
+            Tr.debug(this, tc, "Requesting async write of cache entry");
         }
 
-        try {
-            // delete cache directory if present
-            if (Files.exists(cacheDir)) {
-                if (Files.isDirectory(cacheDir)) {
-                    Files.walkFileTree(cacheDir, RECURSIVE_DELETER);
-                } else {
-                    throw new IOException("Non-directory found in cache location: " + cacheDir.toAbsolutePath());
+        return executor.submit(() -> {
+            try {
+                if (LoggingUtils.isDebugEnabled(tc)) {
+                    Tr.debug(this, tc, "Beginning cache entry write");
                 }
-            }
 
-            // create cache directory
-            Files.createDirectories(cacheDir);
+                // delete cache directory if present
+                if (Files.exists(cacheDir)) {
+                    if (Files.isDirectory(cacheDir)) {
+                        Files.walkFileTree(cacheDir, RECURSIVE_DELETER);
```

Comment:
Is there a safety issue here if someone could somehow (too) freely configure the cache directory location?

Reply:
Don't think so since we get the cache directory from OSGi which allocates a private working directory to each bundle. The class Javadoc should probably be clearer that `CacheEntry` will assume it has ownership of the `cacheDir` and is allowed to delete anything within it.

Reply:
OK